contrib/util/dupscore.cli.php

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/php
+<?php
+
+/**
+ * CLI script to compute patient duplication scores in patient_data.dupscore.
+ * The score is a measure of the likelihood that the patient is a duplicate of
+ * some patient created before it. Optional arguments specifying values are:
+ *
+ * --webdir   The full path to the OpenEMR web directory. Defaults to the directory
+ *            two levels above that of this script.
+ * --site     The site ID. Defaults to "default".
+ * --maxmins  The maximum number of minutes to run. Defaults to 60. Use 0 for no limit.
+ *
+ * Arguments not having a value may be:
+ *
+ * -q         Suppresses messages on stdout.
+ * -c         Clears existing scores to recompute all of them; except scores of -1
+ *            are not cleared because they are manually assigned.
+ *
+ * Because we are comparing every patient with every other patient, this script can
+ * run for a very long time with a large database. Thus we want to do it offline.
+ * If --maxmins is exceeded the script will terminate but may be run again to resume
+ * where it left off.
+ *
+ * A common usage is:
+ * php /var/www/html/openemr/contrib/util/dupscore.cli.php --maxmins=240
+ *
+ * Here is a sample crontab entry to automatically run up to 2 hours nightly:
+ * 3 1 * * * root php /var/www/html/openemr/contrib/util/dupscore.cli.php -q --maxmins=120
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Rod Roark <rod@sunsetsystems.com>
+ * @copyright Copyright (c) 2021 Rod Roark <rod@sunsetsystems.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+// The number of scores to compute between tests for time expiration.
+$querylimit = 1000;
+
+if (php_sapi_name() !== 'cli') {
+    die("This script must be run from the command line!\n");
+}
+
+$args = getopt('cq', array('webdir:', 'site:', 'maxmins:'));
+
+// print_r($args); // debugging
+
+$args['webdir'] = $args['webdir'] ?? dirname(dirname(dirname(__FILE__)));
+$args['site'] = $args['site'] ?? 'default';
+$args['maxmins'] = floatval($args['maxmins'] ?? 60);
+
+if (stripos(PHP_OS, 'WIN') === 0) {
+    $args['webdir'] = str_replace("\\", "/", $args['webdir']);
+}
+
+// Bring in the getDupScoreSQL() function.
+require_once($args['webdir'] . "/library/dupscore.inc.php");
```

Comment:
might as well bring this in after globals.php (ie. standard openemr script organization and then can use a openemr context variable for the path)

Reply:
Done.